### PR TITLE
5214 difference against corrected sections

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
*Issue #:* /bcgov/entity#5214

*Description of changes:*
 - difference against corrected sections

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/reports/report.py [hunk 1]
```diff
@@ -315,6 +320,144 @@ def _format_incorporation_data(self, filing):
...
+    def _format_with_diff_data(self, filing):
+        if incorporation_application := filing['incorporationApplication']:
+            diff = filing.get('correction', {}).get('diff', [])
+
+            self._format_name_translations_with_diff_data(filing, diff)
+            self._format_office_with_diff_data(incorporation_application, diff)
+            self._format_party_with_diff_data(incorporation_application, diff)
+            self._format_share_class_with_diff_data(incorporation_application, diff)
+
+    def _format_name_translations_with_diff_data(self, filing, diff):  # pylint: disable=no-self-use;
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/reports/report.py:347]: This function need to be refactored after (Change Translated Names to managed array#5684)

Comment (thorwolpert) [reply legal-api/src/legal_api/reports/report.py:347]: the change should be there to pick up.

**Review by lekshmimallika-aot — APPROVED** [review]
I dint get a chance to go through this in detail. Overall it looks ok to me.

**Review by thorwolpert — APPROVED** [review]
nice!

**Pull request merged**